engine-tests/src/test/java/org/terasology/documentation/apiScraper/util/ApiMethod.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.documentation.apiScraper.util;
+
+
+public class ApiMethod {
+
+    public static final String DEFAULT_VALUE = "DEFAULT";
+
+    private String className;
+    private String name;
+    private String returnType;
+    private String exceptionType;
+    private String parametersType;
+
+    public ApiMethod(String className, String name, String returnType, String exceptionType, String parametersType){
+        this.className = className;
+        this.name = name;
+        this.returnType = returnType;
+        this.exceptionType = exceptionType;
+        this.parametersType = parametersType;
+    }
+
+    public ApiMethod(){
+        this(DEFAULT_VALUE,DEFAULT_VALUE,DEFAULT_VALUE,DEFAULT_VALUE,DEFAULT_VALUE);
```

Comment:
I am not sure if this constructor (or the default values) is required at all. The only uses that I can see are in the testing methods, and I am sure those can be worked around. Any specific reason why this constructor exists?

Reply:
Yes, there is, but with some refactors I may be able to remove it